### PR TITLE
descriptions: Implement description edit

### LINKED ISSUE #1437: ODP Comments view/edit
client/pages/OriginalDataPoint/components/Comments/CommentsEditor.tsx

### PR DESCRIPTION
**Pull request by sorja**
Resolves #1437

### CODE REVIEW

File: server/controller/cycleData/index.ts [hunk 1]
```diff
@@ -17,4 +17,5 @@ export const CycleDataController = {
   deleteNodeValues,
   getReviewSummary: MessageTopicUserRepository.getReviewSummary,
   getDescription: DescriptionRepository.getOneOrNone,
+  updateDescription: DescriptionRepository.update,
```
Comment (minotogna) [inline server/controller/cycleData/index.ts:20]: We need to persist the activity log as well

File: server/repository/assessmentCycle/descriptions/update.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    sectionName: string
+    name: CommentableDescriptionName
+  },
+  client: BaseProtocol = DB
+): Promise<string> => {
+  const { countryIso, assessment, cycle, sectionName, name, content } = props
+  const schemaCycle = Schemas.getNameCycle(assessment, cycle)
+
+  // insert new | on conflict update
+  const query = `
```
Comment (minotogna) [inline server/repository/assessmentCycle/descriptions/update.ts:24]: Do we need an insert if it doesn't exist? You can handle it in one query with 'on conflict' and we can call the method upsert ?

**Review by minotogna — CHANGES_REQUESTED** [review]

File: server/repository/assessmentCycle/descriptions/update.ts (hunk 2 above)
Comment (sorja) [reply server/repository/assessmentCycle/descriptions/update.ts:24]: Doesn't this do this?

File: meta/assessment/activityLog.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ export enum ActivityLogMessage {
   originalDataPointUpdate = 'originalDataPointUpdate',
   originalDataPointRemove = 'originalDataPointRemove',
   updateCountry = 'updateCountry',
+  updateCountryDescription = 'updateCountryDescription',
```
Comment (minotogna) [inline meta/assessment/activityLog.ts:10]: I would say descriptionUpdate

File: server/api/assessment/index.ts [hunk 4]
```diff
@@ -56,6 +57,7 @@ export const AssessmentApi = {
     express.get(ApiEndPoint.Assessment.sections(), AuthMiddleware.requireView, getSections)
     express.get(ApiEndPoint.Assessment.Sections.Metadata.many(), AuthMiddleware.requireView, getSectionMetadata)
     express.get(ApiEndPoint.Assessment.Data.descriptions(), AuthMiddleware.requireView, getDescription)
+    express.patch(ApiEndPoint.Assessment.Data.descriptions(), AuthMiddleware.requireEdit, upsertDescription)
```
Comment (minotogna) [inline server/api/assessment/index.ts:60]: This is a put

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**